test/integration/suites/upgrade/01-run-upgrade-tests

```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
 
+# TODO: in 1.1.0. Once we're no longer testing 0.12.0, we can and should fix
+# this test and these commands to rely on the default socket path. We can't do
+# it until then because 0.12.0 does not understand the new CLI flag on the
+# server, and also doesn't make the socket directory like the agent (which
+# gives a little needless friction using the new default, since we'd need
+# something else to create the directory first).
```

Comment:
As commented above, we should probably fix the agent thing as part of this change? Is there a reason the agent shouldn't create the dir if it doesn't exist?

Reply:
The agent already does create the directory. It is the server that does not (until this PR) and is what is holding us up from using the default socket path in this upgrade test. Even if it did (and could), we'd still have to specify the socket path on all of these invocations to use that path, so there isn't much to gain.

Reply:
Ooh ok that makes sense, looks like I misread the comment. Thank you for clarifying!